src/main/java/com/mongodb/spark/sql/connector/config/AbstractMongoConfig.java

```diff
@@ -70,6 +69,7 @@ enum UsageMode {
   private final Map<String, String> options;
   private final UsageMode usageMode;
   private transient MongoClientFactory mongoClientFactory;
+  private transient CaseInsensitiveStringMap caseInsensitiveOptions;
```

Comment:
Unfortunately, CaseInsensitiveStringMap is not serializable - so we make it transient